src/libcrun/cgroup.c

```diff
@@ -2100,7 +2172,10 @@ write_hugetlb_resources (int dirfd, bool cgroup2,
       xasprintf (&filename, "hugetlb.%s.%s", htlb[i]->page_size, suffix);
 
       len = sprintf (fmt_buf, "%lu", htlb[i]->limit);
-      ret = write_file_at (dirfd, filename, fmt_buf, len, err);
+      if (cgroup2)
```

Comment:
Can you just put this check into the function, rather then complicating the code?

Reply:
yes, good idea.  I've moved the cgroup2 flag inside the `write_file_and_check_controllers_at` function.